core/exchange_test.go

```diff
@@ -121,6 +121,7 @@ func TestExchange_DoNotStoreHistoric(t *testing.T) {
 func TestExchange_StoreHistoricIfArchival(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	t.Cleanup(cancel)
+	t.Setenv("CELESTIA_OVERRIDE_AVAILABILITY_WINDOW", "1ns") // all blocks will be "historic"
```

Comment:
So this works too but what was not working about the previous approach where we had the `WithAvailabilityWindow(time.Nanosecond)` ?

Reply:
Ah wow yes correct. This test is actually so stupid 😂  Thanks for the catch @chatton !

This test is a false positive. We try to generate non-empty blocks, but non-empty block generation only really begins from height 50 onwards -- we request range (1:30], all are empty blocks, so of course we do not store the Q4. 

This test needs to be completely refactored so that we have a deterministic way to ensure we are querying a range where there are several non-empty blocks included. 

Would you like to do this or should I and I can commit directly to the branch?

Reply:
hey @renaynay thanks for verifying, feel free to push straight to the branch! I think you will have a better idea of how to make sure this is testing the correct behaviour.